vezba-stabla/src/StabloProgram.java

```diff
@@ -0,0 +1,203 @@
+    }
+    
+    protected Stablo(Cvor cvor) {
+    	this.koren = cvor;
+    }
+    
+    
+    // ------ dodati opisane metode ------------
+    
+    // ispisuje registracije svih automobila iz Novog Sada
+    public void ispisiNS() {
+    	System.out.println("Svi automobili iz Novog Sada: ");
+    	sviIzNs(koren);
+    }
+    
+    private void sviIzNs(Cvor cvor) {
+    	if (cvor == null) 
+    		return;
+    	
+    	if (cvor.info.getRegistracija().contains("NS")) {
+    		System.out.println(cvor.info.getRegistracija());
+    	}
+    	
+    	sviIzNs(cvor.levo);
+    	sviIzNs(cvor.desno);    	
+    }
+    
+    // vraca sumu kilometraza svih plavih automobila
+    public long presliPlavi() {
+    	Suma suma = new Suma();
+    	presliPlavi(koren, suma);
+    	System.out.println("Plavi automobili su presli: ");
+    	return suma.getSuma();
+    }
+    
+    private void presliPlavi(Cvor cvor, Suma suma) {
+    	if (cvor == null)
+    		return;
+    	
+    	if (cvor.info.getBoja().equals("Plava")) {
+    		suma.setSuma(cvor.info.getKilometraza());
+    	}
+    	
+    	presliPlavi(cvor.levo, suma);
+    	presliPlavi(cvor.desno, suma);
+    }
+    
+
+    /*
+     * Pronalazi u stablu auto sa prosledjenom registracijom i
+     * kreira novo stablo u kojem pronadjeni auto ima za @param long km
+     * povecanu predjenu kilometrazu
+     */
+    public Stablo presaoJos(String registracija, long km) {
+    	Cvor korenNovog = kopijaCvora(koren, registracija, km);
+    	return new Stablo(korenNovog);
+    }
+    
+    private Cvor kopijaCvora(Cvor cvor, String reg, long km) {
+    	if (cvor == null) 
+    		return null;
+    	
+    	Cvor novi;
+    	if (cvor.info.getRegistracija().equals(reg)) {
+    		Automobil auto = new Automobil(cvor.info.getModel(), 
+    										cvor.info.getBoja(), 
+    											reg, (cvor.info.getKilometraza() + km));
+    		novi = new Cvor(auto, null, null);
+    	} else {
+    		novi = new Cvor(cvor.info, null, null);
+    	}
+    	 
+    	novi.levo = kopijaCvora(cvor.levo, reg, km);
+    	novi.desno = kopijaCvora(cvor.desno, reg, km);
+    	return novi;
+    }
+    
+    /*
+     * Vraca listu automobila koji imaju vecu kilometrazu od 
+     * svog roditelja i susednog cvora (bratskog)
+     */
+    public List<Automobil> sortiraniPlavi() {
+    	List<Automobil> lista = new ArrayList<Automobil>();
+    	listaPlavih(lista, koren, null);
+    	return lista;
+    }
+    
+    private void listaPlavih(List<Automobil> lista, Cvor tekuci, Cvor roditelj) {
+    	if (tekuci == null) 
+    		return;
+    	
+    	// prvo proverimo da li je auto plavi i da li ima roditelja
+    	if (tekuci.info.getBoja().equals("Plava") && roditelj != null) {
+    		// zatim proverimo da li tekuci cvor ima vecu kilometrazu
+    		// od roditelja ako nema onda ne proveravamo nista dalje
+    		if (tekuci.info.getKilometraza() > roditelj.info.getKilometraza()) {
+    			// gledamo da li nas tekuci cvor ima susednog i da li susedni zadovoljava uslov

```

Comment:
Super je samo bih voleo da se objasni zašto neće dodati u listu ako je jedini sin. Recimo ako je tekući levi čvor i sad proverimo levog, onda on neće biti dodat u listu jer levi (tekući) nije manji od tekućeg (samog sebe). To možda samo naglasiti da ne bi moralo da se mozga kad se čita.